Source/Core/DolphinQt/Config/Mapping/MappingNumeric.cpp

```diff
@@ -7,26 +7,54 @@
 #include "DolphinQt/Config/Mapping/MappingWidget.h"
 
 #include "InputCommon/ControllerEmu/ControllerEmu.h"
-#include "InputCommon/ControllerEmu/Setting/NumericSetting.h"
 #include "InputCommon/ControllerInterface/ControllerInterface.h"
 
-MappingNumeric::MappingNumeric(MappingWidget* parent, ControllerEmu::NumericSetting* setting)
-    : m_setting(*setting)
+MappingDouble::MappingDouble(MappingWidget* parent, ControllerEmu::NumericSetting<double>* setting)
+    : QDoubleSpinBox(parent), m_setting(*setting)
 {
-  setRange(setting->m_low, setting->m_high);
+  setRange(m_setting.GetMinValue(), m_setting.GetMaxValue());
 
-  connect(this, static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged), this,
-          [this, parent](int value) {
-            m_setting.SetValue(static_cast<double>(value) / 100);
+  setDecimals(2);
+
+  const auto suffix = tr(m_setting.GetUISuffix());
+  if (!suffix.isEmpty())
+    setSuffix(QStringLiteral(" ") + suffix);
+
+  connect(this, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this,
+          [this, parent](double value) {
+            m_setting.SetValue(value);
             parent->SaveSettings();
           });
 
-  connect(parent, &MappingWidget::ConfigChanged, this, &MappingNumeric::ConfigChanged);
+  connect(parent, &MappingWidget::ConfigChanged, this, &MappingDouble::ConfigChanged);
+}
+
+void MappingDouble::fixup(QString& input) const
```

Comment:
I don't see this being used, but if you keep it; we usually pass parameters that are intended to be modified as pointer (and not as reference).

Reply:
That's an overridden function from qt. I didn't choose the signature. :P